instrumentation/apolloconfig-apolloclient-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apolloconfig/apolloclient/v1_0/ApolloConfigSingletons.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apolloconfig.apolloclient.v1_0;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanKindExtractor;
+import javax.annotation.Nullable;
+
+public final class ApolloConfigSingletons {
+
+  private static final String INSTRUMENTATION_NAME =
+      "io.opentelemetry.apolloconfig-apolloclient-1.0";
+  private static final Instrumenter<String, Void> INSTRUMENTER;
+
+  private static final AttributeKey<String> CONFIG_NS_ATTRIBUTE_KEY = stringKey("config.namespace");
+  public static final ContextKey<String> REPOSITORY_CHANGE_REPEAT_CONTEXT_KEY =
+      ContextKey.named("apollo-config-repository-change-repeat");
+
+  static {
+    AttributesExtractor<String, Void> attributesExtractor =
+        new AttributesExtractor<String, Void>() {
+
+          @Override
+          public void onStart(
+              AttributesBuilder attributes, Context parentContext, String namespace) {
+            if (namespace == null) {
+              return;
+            }
+
+            attributes.put(CONFIG_NS_ATTRIBUTE_KEY, namespace);
+          }
+
+          @Override
+          public void onEnd(
+              AttributesBuilder attributes,
+              Context context,
+              String namespace,
+              @Nullable Void unused,
+              @Nullable Throwable error) {}
+        };
+
+    INSTRUMENTER =
+        Instrumenter.<String, Void>builder(
+                GlobalOpenTelemetry.get(),
+                INSTRUMENTATION_NAME,
+                (event) -> "Apollo Config Repository Change")
+            .setSpanStatusExtractor(
+                (spanStatusBuilder, request, unused, error) -> {
+                  if (error != null) {
+                    spanStatusBuilder.setStatus(StatusCode.ERROR);
+                  }
+                })
+            .addAttributesExtractor(attributesExtractor)
+            .buildInstrumenter(SpanKindExtractor.alwaysClient());
```

Comment:
Have a look at https://opentelemetry.io/docs/concepts/signals/traces/#client As far as I can tell this span does not behave as a client span as outlined in the specification.

Reply:
Based on our previous discussion, this is indeed not a traditional client type, but more like a job consumer for configuration changes. I will change it to a Consumer type.

![image](https://github.com/user-attachments/assets/9b1f919b-6951-43ac-afee-22f29f86881a)

Reply:
> Based on our previous discussion, this is indeed not a traditional client type, but more like a job consumer for configuration changes. I will change it to a Consumer type.

Consumer span is intended to be used for consuming asynchronous messages e.g kafka, jms. Server span are for synchronous calls e.g. http. You probably should make it an `INTERNAL` span as consumer doesn't really suite for it either.